### PR TITLE
feat: implement spanner_create_storing_index example

### LINKED ISSUE #480: Implement spanner_create_storing_index example.
Use the admin API to update the DDL with a new index with the `STORING` directive:

https://github.com/GoogleCloudPlatform/golang-samples/blob/40a33f7ff5de11da00a6715eaa29ce45fdf92387/spanner/spanner_snippets/snippet.go#L696-L715

### PR DESCRIPTION
**Pull request by tmatsuo**
fixes #480

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/1029)

### CODE REVIEW

File: google/cloud/spanner/samples/samples.cc [hunk 1]
```diff
@@ -529,6 +529,36 @@ void AddColumn(std::vector<std::string> const& argv) {
...
+    google::cloud::spanner::DatabaseAdminClient client;
+    google::cloud::spanner::Database database(project_id, instance_id,
+                                              database_id);
+    future<StatusOr<
+        google::spanner::admin::database::v1::UpdateDatabaseDdlMetadata>>
+        future = client.UpdateDatabase(database, {R"""(
+            CREATE INDEX AlbumsByAlbumTitle2 ON Albums(AlbumTitle)
+                STORING (MarketingBudget))"""});
+    StatusOr<google::spanner::admin::database::v1::UpdateDatabaseDdlMetadata>
+        result = future.get();
```
Comment (devjgm) [inline google/cloud/spanner/samples/samples.cc]: Nit: I think it's nice to name these StatusOr variables after the type/object that they (might) contain. Like in this case, maybe name it `metadata`. That way, the rest of the code reads nicely as if the caller has a (smart)pointer to a metadata object, which they kinda do.

Comment (tmatsuo) [reply google/cloud/spanner/samples/samples.cc]: Thanks, renamed the variable. I'll merge when green

**Comment by tmatsuo** [comment]
Windows CMake build failed 3 times in a row with segfault.

**Comment by tmatsuo** [comment]
I'll re-run the test again and try to monitor the VM which runs the test.

**Comment by tmatsuo** [comment]
This time samples passes, but we got:
51 - instance_admin_integration_test (Timeout)

**Pull request merged**